hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -676,6 +677,60 @@
 
 
     /**
+     * Type of Query Index result copying behavior.
+     *
+     * Defines the behavior for index copying on index read/write.
+     *
+     * Supported in BINARY, OBJECT and NATIVE in-memory-formats.
+     *
+     * Why is it needed? In order to support correctness the internal data-structures used by indexes need to do some copying.
+     * The copying may take place on-read or on-write:
+     *
+     * -> Copying on-read means that each index-read operation will copy the result of the query before returning it to the
+     * caller.This copying may be expensive, depending on the size of the result, since the result is stored in a map, which
+     * means that all entries need to have the hash calculated before being stored in a bucket.
+     * Each index-write operation however will be fast, since there will be no copying taking place.
+     *
+     * -> Copying on-write means that each index-write operation will completely copy the underlying map to provide the
+     * copy-on-write semantics. Depending on the index size, it may be a very expensive operation.
+     * Each index-read operation will be very fast, however, since it may just access the map and return it to the caller.
+     *
+     * -> Never copying is tricky. It means that the internal data structures of the index are concurrently modified without
```

Comment:
......tricky. It means that the underlying data structures used by the index are....